src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/ReportedAdjudicationServiceTest.kt

```diff
@@ -274,6 +327,69 @@ class ReportedAdjudicationServiceTest {
     }
   }
 
+  @Nested
+  inner class ReportedAdjudicationSetStatus {
+    private fun reportedAdjudication(): ReportedAdjudication {
+      val reportedAdjudication = ReportedAdjudication(
+        reportNumber = 123, prisonerNumber = "AA1234A", bookingId = 123, agencyId = "MDI",
+        dateTimeOfIncident = DATE_TIME_OF_INCIDENT, locationId = 345, statement = INCIDENT_STATEMENT,
+        incidentRoleCode = "25b", incidentRoleAssociatedPrisonersNumber = "BB2345B",
+        offenceDetails = mutableListOf(
+          ReportedOffence(
+            offenceCode = 3,
+            paragraphCode = OFFENCE_CODE_3_PARAGRAPH_CODE,
+            victimPrisonersNumber = "BB2345B",
+            victimStaffUsername = "DEF34G",
+            victimOtherPersonsName = "Another Name",
+          )
+        ),
+        handoverDeadline = DATE_TIME_REPORTED_ADJUDICATION_EXPIRES,
+        status = ReportedAdjudicationStatus.AWAITING_REVIEW,
+        statusReason = null,
+        statusDetails = null,
+      )
+      reportedAdjudication.createdByUserId = "A_SMITH"
+      reportedAdjudication.createDateTime = REPORTED_DATE_TIME
+      return reportedAdjudication
+    }
+
+    private val allTransitions =
+      ReportedAdjudicationStatus.values().map { from -> ReportedAdjudicationStatus.values().map { to -> from to to } }
+        .flatten()
+    private val validTransitions =
+      ReportedAdjudicationStatus.values().map { from -> from.nextStates().map { to -> from to to } }.flatten()
+    private val invalidTransitions = allTransitions.minus(validTransitions)
+
+    @Test
+    fun `setting status for a reported adjudication throws an illegal state exception for invalid transitions`() {
```

Comment:
I think this is a perfect place to use parameterised tests.

I don't think we need to test all transitions like this - the main problem is that it is using the code to check the code. So if the code is wrong then it will still pass.

By using parameterised tests you can cover some (or all) scenarios without relying on the code itself. Obviously the down side to that is the usual downside for duplication - you have 2 places to change if it is wrong, but I think it would make the tests a bit more representative.

Reply:
will do!